docker/compute_worker/compute_worker.py

```diff
@@ -445,15 +443,20 @@ def start(self):
         ingestion_program_dir = os.path.join(self.root_dir, "ingestion_program")
 
         logger.info("Running scoring program, and then ingestion program")
-        self._run_program_directory(program_dir, kind='program', can_be_output=True)
-        self._run_program_directory(ingestion_program_dir, kind='ingestion')
-
-        # Unpack submission and data into some directory
-        # Download docker image
-        # ** When running SCORING PROGRAM ** pass by volume the codalab.py library file so submissions/organizers can use it
-        # Normal things pass all run_args as env vars to submission
-        # Upload submission results
-        # Upload submission stdout/etc.
+
+        if self.ingestion_only_during_scoring and self.is_scoring:
+            asyncio.get_event_loop()
+            asyncio.get_child_watcher()
+            thread1 = threading.Thread(target=self._run_program_directory, args=(program_dir, 'program', True))
```

Comment:
nit picking, we could name this `program_thread` and `ingestion_program_thread` maybe?